api/src/neo4j/queries/search.js

```diff
@@ -351,13 +352,33 @@ LIMIT ${limit}
     fetchCompartments({ ids: ids["Compartment"], model, version: v, includeCounts: true }),
   ]);
 
-  // formatting for simple (gem browser) search
+  const resObj = {compartmentalizedMetabolites,
+    metabolites,
+    genes,
+    reactions,
+    subsystems,
+    compartments}
+    const resWithScore = {};
+    for (const [component, result] of Object.entries(resObj)) {
+        if (result) {
+            resWithScore[component] = result.map(obj => {
+                    const objWithScore = {
+                        ...obj,
+                        score: idsToScore[obj.id]
+                    }
+                    return objWithScore;
+                })
+        } else {
+            resWithScore[component] = []
+        }
+    }
```

Comment:
I think a linter would suggest some changes here:

```suggestion
  const resObj = {
    compartmentalizedMetabolites,
    metabolites,
    genes,
    reactions,
    subsystems,
    compartments
  };

  const resWithScore = {};
  for (const [component, result] of Object.entries(resObj)) {
    if (result) {
      resWithScore[component] = result.map(obj => {
        const objWithScore = {
          ...obj,
          score: idsToScore[obj.id]
        };
        return objWithScore;
      });
    } else {
      resWithScore[component] = [];
    }
  }
```